configs/common/packages.yaml

```diff
@@ -243,10 +243,10 @@ packages:
   # https://github.com/JCSDA/spack-stack/issues/1276
   py-matplotlib:
     require: '@3.7.4'
-  # Variant ~mpi not working for latest py-netcdf4
-  # https://github.com/spack/spack/issues/47652
+  # https://github.com/Unidata/netcdf4-python/issues/1389
+  # https://github.com/Unidata/netcdf4-python/issues/1389#issuecomment-2599760051
   py-netcdf4:
-    require: '@1.7.1 +mpi'
```

Comment:
This change is required to fix an error with `py-netcdf4` on certain systems when built with `+mpi`. We used to build `py-netcdf4` without `mpi`, but for a period this wasn't possible until we added a patch to disable the py-netcdf auto-detect feature. That patch allows us to build `py-netcdf4 ~mpi` even if `netcdf-c` was built with `+mpi`.

WIP